packages/svelte/src/main/main-client.js

```diff
@@ -25,14 +28,11 @@ import { is_array } from '../internal/client/utils.js';
  * @returns {void}
  */
 export function onMount(fn) {
-	if (!is_ssr) {
-		user_effect(() => {
-			const result = untrack(fn);
-			if (typeof result === 'function') {
-				return /** @type {() => any} */ (result);
-			}
-		});
+	if (current_component_context === null) {
+		throw new Error('onMount can only be used during component initialisation.');
 	}
+
+	(current_component_context.u ??= init_update_callbacks(current_component_context)).m.push(fn);
```

Comment:
I'm very much against disallowing `onMount` (or deprecating/removing it) because it will just mean that everyone will have their own version of it in some `utils` folder. I think it's better to provide it ourselves, and differentiate how to invoke it depending on whether or not you're in runes mode.
Disallowing `beforeUpdate/afterUpdate` in runes mode makes sense to me, given that it's not used often and probably pretty straightforward to migrate in 95% of all cases.

Reply:
Yeah, I wasn't proposing disallowing `onMount`, just the other two. I'll make that change